### PR TITLE
Stop river from starting duplicate services

### PR DESCRIPTION
**Pull request by ghost**
fixed version of other commit that got ruined by build.yml updates from me trying to build an image with GH actions to test

### CODE REVIEW

**Review by rehashedsalt — APPROVED** [review]
Changes look good to me. Per previous discussion in #79, systemd spawns a lot of services per user, including pipewire.

To be clear: is screensharing on Wayblue River broken right now out-of-the-box?

**Comment by ghost** [comment]
using the default river config on my secureblue install, yes. Because it spawns pipewire manually which conflicts with the one systemd starts automaticall

**Comment by ghost** [comment]
i can do a quick test in a VM if you want

**Comment by rehashedsalt** [comment]
Extra testing is always a good idea. I'm not familiar with River so much as I am Hyprland, so it'd take me longer to maneuver around and reproduce the issue.

File: files/system/river/etc/river/init [hunk 1]
```diff
@@ -170,13 +170,11 @@ riverctl map normal Super Escape spawn "swaylock -f"
...
 
 riverctl spawn dunst
-riverctl spawn pipewire
-riverctl spawn pipewire-pulse
-riverctl spawn wireplumber
 riverctl spawn "/usr/libexec/xfce-polkit"
 riverctl spawn waybar
 riverctl spawn kanshi
 riverctl spawn nm-applet
+riverctl spawn "swaybg --image /usr/share/backgrounds/f41/default/f41-01-night.png"
```
Comment (RoyalOughtness) [inline files/system/river/etc/river/init]: should be `/usr/share/backgrounds/default.png` I believe

Comment (ghost) [reply files/system/river/etc/river/init]: ok, will fix

**Comment by ghost** [comment]
i haven't figured out building a test image yet but i fixed the default wallpaper path to `/usr/share/backgrounds/default.png`

**Comment by RoyalOughtness** [comment]
> i haven't figured out building a test image yet but i fixed the default wallpaper path to `/usr/share/backgrounds/default.png`

one way is in your fork if you swap out the cosign keys, you can build in github actions and rebase to your own fork

**Comment by ghost** [comment]
> > i haven't figured out building a test image yet but i fixed the default wallpaper path to `/usr/share/backgrounds/default.png`
> 
> one way is in your fork if you swap out the cosign keys, you can build in github actions and rebase to your own fork

audio,screensharing and wallpaper doesnt work on the currently latest wayblue image on a fresh user
i have built my image and about to reboot to it but it installed build tools aswell
`Added:
  binutils-2.43.1-2.fc41.x86_64
  debian-keyring-2023.4-4.fc41.noarch
  gcc-14.2.1-3.fc41.x86_64
  gcc-c++-14.2.1-3.fc41.x86_64
  glibc-devel-2.40-9.fc41.x86_64
  kernel-headers-6.11.3-300.fc41.x86_64
  libstdc++-devel-14.2.1-3.fc41.x86_64
  libxcrypt-devel-4.4.36-10.fc41.x86_64
  make-1:4.4.1-8.fc41.x86_64
  nodejs-1:22.11.0-1.fc41.x86_64
  nodejs-docs-1:22.11.0-1.fc41.noarch
  nodejs-full-i18n-1:22.11.0-1.fc41.x86_64
  nodejs-libs-1:22.11.0-1.fc41.x86_64
  nodejs-npm-1:10.9.0-1.22.11.0.1.fc41.x86_64
  ripgrep-14.1.1-1.fc41.x86_64
  swaybg-1.2.1-2.fc41.x86_64
  tree-sitter-cli-0.23.2-1.fc41.x86_64
  ubu-keyring-2023.11.28.1-3.fc41.noarch
  xsel-1.2.1-2.fc41.x86_64Added:
  binutils-2.43.1-2.fc41.x86_64
  debian-keyring-2023.4-4.fc41.noarch
  gcc-14.2.1-3.fc41.x86_64
  gcc-c++-14.2.1-3.fc41.x86_64
  glibc-devel-2.40-9.fc41.x86_64
  kernel-headers-6.11.3-300.fc41.x86_64
  libstdc++-devel-14.2.1-3.fc41.x86_64
  libxcrypt-devel-4.4.36-10.fc41.x86_64
  make-1:4.4.1-8.fc41.x86_64
  nodejs-1:22.11.0-1.fc41.x86_64
  nodejs-docs-1:22.11.0-1.fc41.noarch
  nodejs-full-i18n-1:22.11.0-1.fc41.x86_64
  nodejs-libs-1:22.11.0-1.fc41.x86_64
  nodejs-npm-1:10.9.0-1.22.11.0.1.fc41.x86_64
  ripgrep-14.1.1-1.fc41.x86_64
  swaybg-1.2.1-2.fc41.x86_64
  tree-sitter-cli-0.23.2-1.fc41.x86_64
  ubu-keyring-2023.11.28.1-3.fc41.noarch
  xsel-1.2.1-2.fc41.x86_64`

i only have these packages layered

`arch-install-scripts btop cascadia-code-nf-fonts cloudflare-warp debootstrap gparted htop light lxqt-policykit neovim qemu qemu-system-x86`

**Comment by ghost** [comment]
on my image both wallpaper and screensharing works, perhaps documentation could be added on how to customize wallpaper

**Comment by ghost** [comment]
the process of contributing to a ublue image is really download-intensive, this will make it exponentially harder to contribute the slower a user's download speed is, i am lucky to have a fast download speed

File: files/system/river/etc/river/init (hunk 1 above)
Comment (ghost) [reply files/system/river/etc/river/init]: it does show /usr/share/backgrounds/default.png on a fresh user, resolving conversation

**Comment by RoyalOughtness** [comment]
> currently latest wayblue image
you mean specifically on the river image right? not all of them?

**Comment by RoyalOughtness** [comment]
> debootstrap
> arch-install-scripts

this is tangential but you can just run `ujust distrobox-assemble` and make arch and debian distroboxes instead

**Comment by ghost** [comment]
> > currently latest wayblue image
> > you mean specifically on the river image right? not all of them?

i was only concerned with the river image, i can fix these 2 issues on all images that have it if you want

**Comment by ghost** [comment]
i haven't checked any other image

**Comment by ghost** [comment]
my internet is probably not fast enough to try all images in 2 hours

**Comment by RoyalOughtness** [comment]
>  can fix these 2 issues on all images that have it if you want

Last I checked, other images don't have this issue

**Comment by ghost** [comment]
> > can fix these 2 issues on all images that have it if you want
> 
> Last I checked, other images don't have this issue

i havent tested them either and they probably dont have this issue, if its fine with you we can merge

**Review by rehashedsalt — APPROVED** [review]

**Comment by ghost** [comment]
after messing up the previous PR i wont touch a branch i submited a PR for and seperate each commit into a seperate branch to PR next time

**Review by ghost — COMMENTED** [review]
i am running these changes right now and they both work

**Comment by rehashedsalt** [comment]
I don't think default Hyprland has any problems, but I'm removing the portal invocations from its config anyway just to remove variables.

It's tough to get reports for this sort of thing because the target audience for Wayblue is very likely to rip out the default configs and build their own, addressing issues themselves or going to circles dedicated to their compositor.

But that's tangential. Changes in this PR lgtm

**Comment by ghost** [comment]
sorry, i am pretty new to github

**Comment by ghost** [comment]
im lucky that swaybg is one of the [recommended wallpaper programs](https://codeberg.org/river/wiki/src/branch/master/pages/Recommended-Software.md#wallpaper-tools) by upstream otherwise other people could have asked for their preferred wallpaper program also one of the goals of wayblue is to be slightly opinionated and if someone prefers hyprpaper they could find swaybg unececarry

edit: swaybg is what i happened to use on my river config

**Comment by ghost** [comment]
so it may make more sense to leave riverwm wallpaperless and just tell people to layer their preferred wallpaper program on top of it

**Comment by ghost** [comment]
> I don't think default Hyprland has any problems, but I'm removing the portal invocations from its config anyway just to remove variables.
> 
> It's tough to get reports for this sort of thing because the target audience for Wayblue is very likely to rip out the default configs and build their own, addressing issues themselves or going to circles dedicated to their compositor.
> 
> But that's tangential. Changes in this PR lgtm

hmm i need to check if xdg-desktop-portal-hyprland is automatically started by systemd because when i was testing the previous PR it seemed like xdg-desktop-portal-wlr would be somehow started automatically by xdg-desktop-portal without it being enabled as a systemd user service but not xdg-desktop-portal-hyprland meaning removing the lines from hyprland could break screensharing there
 
edit: The PR may look good but still needs more testing, i dont have much free time now

**Comment by rehashedsalt** [comment]
To be clear: the first thing I did when I switched to Wayblue hyprland about a month and change ago was rip out the default configuration for my own, which does not start any portals nor Pipewire. I've had no issues just letting systemd spawn them as part of the graphical session:

```
$ systemctl -t service --user --no-legend --no-pager --state=running
[...]
  pipewire-pulse.service                                loaded active running PipeWire PulseAudio
  pipewire.service                                      loaded active running PipeWire Multimedia Service
  wireplumber.service                                   loaded active running Multimedia Service Session Manager
  xdg-desktop-portal-gtk.service                        loaded active running Portal service (GTK/GNOME implementation)
  xdg-desktop-portal-hyprland.service                   loaded active running Portal service (Hyprland implementation)
  xdg-desktop-portal.service                            loaded active running Portal service
```

**Comment by ghost** [comment]
check for the hyprland desktop portal like this:
`$ systemctl list-dependencies --user graphical-session.target´

**Comment by ghost** [comment]
> To be clear: the first thing I did when I switched to Wayblue hyprland about a month and change ago was rip out the default configuration for my own, which does not start any portals nor Pipewire. I've had no issues just letting systemd spawn them as part of the graphical session:
> 
> ```
> $ systemctl -t service --user --no-legend --no-pager --state=running
> [...]
>   pipewire-pulse.service                                loaded active running PipeWire PulseAudio
>   pipewire.service                                      loaded active running PipeWire Multimedia Service
>   wireplumber.service                                   loaded active running Multimedia Service Session Manager
>   xdg-desktop-portal-gtk.service                        loaded active running Portal service (GTK/GNOME implementation)
>   xdg-desktop-portal-hyprland.service                   loaded active running Portal service (Hyprland implementation)
>   xdg-desktop-portal.service                            loaded active running Portal service
> ```

i probably noticed this issue because i was using various distros without systemd and my river config had to start pipewire and the portals manually because most alternate init systems dont have a mature way of managing user services 

edit: for example gentoo's openrc-navi does let me start services on login which works for pipewire but i haven't noticed a way it lets me start some service on graphical login and stop it on logout

**Comment by ghost** [comment]
@RoyalOughtness anything that needs fixing before merge?

**Comment by ghost** [comment]
i will increase quality simillar to #81 and follow it also disabling dunst but commented out

**Comment by ghost** [comment]
wallpapers can be discussed later, i have made the commenting out of duplicate services like #81 with explanation and didnt force push the other commit, i am sorry for being hard to deal with as a contributor

**Comment by ghost** [comment]
updated explanation to match #81 for consistiency

**Pull request merged**